types/ajv-pack/index.d.ts

```diff
@@ -0,0 +1,46 @@
+// Type definitions for ajv-pack 0.3
+// Project: https://github.com/epoberezkin/ajv-pack
+// Definitions by: Mihail Malo <https://github.com/qm3ster>
+//                 Christian Murphy <https://github.com/ChristianMurphy>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+import { Ajv, ValidateFunction } from 'ajv';
+
+interface AjvPack {
+    /**
+     * validate data against the schema
+     * @param schema JSON-schema
+     * @param data data to validate
+     * @return validation result
+     */
+    validate(this: Ajv, schema: object, data: any): boolean;
+
+    /**
+     * compile the schema and require the module
+     * @param  schema JSON-schema
+     * @return validation function
+     */
+    compile(this: Ajv, schema: object): ValidateFunction;
+
+    /**
+     * add schema to the instance
+     * @return result from ajv instance
+     */
+    addSchema: Ajv['addSchema'];
+
+    /**
+     * add custom keyword to the instance
+     * @return result from ajv instance
+     */
+    addKeyword: Ajv['addKeyword'];
+}
+
+interface PackValidate {
+    (ajv: Ajv, validate: ValidateFunction): string;
+    instance(ajv: Ajv): AjvPack;
+}
+
+declare const pack: PackValidate;
```

Comment:
It seems like this is a callable module with a function property `instance`.
For clarity, I would recommend using [this template](http://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-function-d-ts.html) instead of what you have currently.

Reply:
Thanks for the suggestion @uniqueiniquity!
restructured to use a namespace in db88f98487fcd84fd43fe10141b3fd236f2e359f